src/main.ts

```diff
@@ -17,6 +17,11 @@ async function run() {
 
     if(version) {
       await rustup.install();
+      if (os.platform() !== 'darwin') {
+        // update the GitHub managed VM version of rustup
+        // to leverage newer features like "latest latest compatible nightly"
+        await exec.exec('rustup', ['self', 'update']);
+      }
       await exec.exec('rustup', ['default', version]);
```

Comment:
Instead of running `rustup component add` and `rustup target add` after toolchain installation, we can probably issue a `toolchain install` with all the components and targets at once, right here before the `default` command.